hydra-node/src/Hydra/Chain/Direct.hs

```diff
@@ -207,21 +218,19 @@ instance Exception ConnectException
 client ::
   (MonadST m, MonadTimer m) =>
   Tracer m DirectChainLog ->
-  NetworkMagic ->
   TQueue m (ValidatedTx Era) ->
-  Party ->
-  TVar m OnChainHeadState ->
+  TVar m SomeOnChainHeadState ->
   ChainCallback Tx m ->
   NodeToClientVersion ->
   OuroborosApplication 'InitiatorMode LocalAddress LByteString m () Void
-client tracer networkMagic queue party headState callback nodeToClientV =
+client tracer queue headState callback nodeToClientV =
   nodeToClientProtocols
     ( const $
         pure $
           NodeToClientProtocols
             { localChainSyncProtocol =
                 InitiatorProtocolOnly $
-                  let peer = chainSyncClientPeer $ chainSyncClient tracer networkMagic callback party headState
+                  let peer = chainSyncClientPeer $ chainSyncClient tracer callback headState
```

Comment:
Where did the `networkMagic` and `party` go? The latter was required to `observeInitTx` iirc

Reply:
They're now in the state itself.